geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -245,45 +244,61 @@ static int setOpStoreResult(RegionProvider regionProvider, RedisKey destinationK
     return scanResult;
   }
 
-  public Collection<byte[]> spop(Region<RedisKey, RedisData> region, RedisKey key, int popCount) {
-    int originalSize = scard();
-    if (originalSize == 0) {
-      return emptyList();
+  public List<byte[]> spop(int count, Region<RedisKey, RedisData> region, RedisKey key) {
+    int randMethodRatio = 3;
```

Comment:
Can this just be a static constant? Please also add some comments as to what the reasoning is.

I'm somewhat hesitant to add optimizations like this without any quantifying metrics to justify; since, on the face of it, it just ends up complicating the code - especially for someone who may look at it in the future without any context. This is just a general comment and not a request to change it... :)